rule/ipset_linux.go

```
@@ -0,0 +1,66 @@
+//go:build linux
+
+package rules
+
+import (
+	C "github.com/Dreamacro/clash/constant"
+	"github.com/Dreamacro/clash/log"
+	"github.com/vishvananda/netlink"
+)
+
+func init() {
+	ver, minorVer, err := netlink.IpsetProtocol()
+	if err != nil {
+		log.Warnln("ipset protocol failed: %s", err.Error())
+		return
+	}
+	log.Infoln("ipset version: %d.%d", ver, minorVer)
+}
+
+type IPSet struct {
+	name    string
+	adapter string
+}
+
+func (f *IPSet) RuleType() C.RuleType {
+	return C.IPSet
+}
+
+func (f *IPSet) Match(metadata *C.Metadata) bool {
+	exist, err := netlink.IpsetTest(f.name, &netlink.IPSetEntry{
+		IP: metadata.DstIP,
+	})
+	if err != nil {
+		log.Warnln("check ipset '%s' failed: %s", f.name, err.Error())
+		return false
+	}
+	return exist
+}
+
+func (f *IPSet) Adapter() string {
+	return f.adapter
+}
+
+func (f *IPSet) Payload() string {
+	return f.name
+}
+
+func (f *IPSet) ShouldResolveIP() bool {
+	return false
+}
+
+func (f *IPSet) ShouldFindProcess() bool {
+	return false
+}
+
+func NewIPSet(name string, adapter string) *IPSet {
+	_, err := netlink.IpsetList(name)
+	if err != nil {
```

why not just return an error?